hazelcast/src/main/java/com/hazelcast/util/PhoneHome.java

```diff
@@ -253,34 +258,36 @@ private void addManCenterInfo(Node hazelcastNode, int clusterSize, PhoneHomePara
             }
             connection.setConnectTimeout(CONNECTION_TIMEOUT_MILLIS);
             connection.setReadTimeout(CONNECTION_TIMEOUT_MILLIS);
+
             // if management center is not running,
             // connection.getInputStream() throws 'java.net.ConnectException: Connection refused'
             inputStream = connection.getInputStream();
-            responseCode = connection.getResponseCode();
             reader = new InputStreamReader(inputStream, "UTF-8");
-            JsonObject mcPhoneHomeInfoJson = JsonValue.readFrom(reader).asObject();
-            version = JsonUtil.getString(mcPhoneHomeInfoJson, "mcVersion");
-            license = JsonUtil.getString(mcPhoneHomeInfoJson, "mcLicense", null);
+            JsonObject mcPhoneHomeInfoJson = Json.parse(reader).asObject();
+            String version = getString(mcPhoneHomeInfoJson, "mcVersion");
+            String license = getString(mcPhoneHomeInfoJson, "mcLicense", null);
+
+            int responseCode = connection.getResponseCode();
+            if (responseCode == HttpURLConnection.HTTP_OK) {
```

Comment:
There is a behavioural changes so these changes affect meaning of the `mcver` and `mclicense` infos for the marketing team. You can find detailed informations about statuses at related PRD:
https://hazelcast.atlassian.net/wiki/spaces/PM/pages/227148329/Improve+Phone+Home+Data+for+ManCenter+Licenses+TDD

Reply:
I've reverted the order. That part was indeed untested, so we cannot very the exact behavior on different error cases.

Good catch!